nifi-framework-api/src/main/java/org/apache/nifi/controller/queue/FlowFileQueue.java

```diff
@@ -93,6 +93,18 @@
 
     QueueSize size();
 
+    /**
+     * @param fromTimestamp The timestamp in milliseconds from which to calculate durations. This will typically be the current timestamp.
+     * @return the sum in milliseconds of how long all FlowFiles within this queue have currently been in this queue.
+     */
+    long getTotalActiveQueuedDuration(long fromTimestamp);
+
+    /**
+     * @param fromTimestamp The timestamp in milliseconds from which to calculate durations. This will typically be the current timestamp.
+     * @return the maximum time in milliseconds that any FlowFile within this queue has been in this queue.
+     */
+    long getMaxActiveQueuedDuration(long fromTimestamp);
```

Comment:
I'm not sure what exactly it means for the max active queued duration to take into account a timestamp... it makes a lot more sense to me to accept no arguments here at all, and to instead just use the current time. Or, alternatively, to return the timestamp of the longest-queued FlowFile.

Reply:
It is definitely more efficient to return the timestamp and do one calculation vs calculating the difference for every flow file prior to returning a max. I will update accordingly.